tests/utils/environment.go

```diff
@@ -71,19 +72,30 @@ type TestingEnvironment struct {
 	Scheme             *runtime.Scheme
 	PreserveNamespaces []string
 	Log                logr.Logger
+	PostgresVersion    int
 }
 
 // NewTestingEnvironment creates the environment for testing
 func NewTestingEnvironment() (*TestingEnvironment, error) {
 	var env TestingEnvironment
+	var err error
 	env.RestClientConfig = ctrl.GetConfigOrDie()
 	env.Interface = kubernetes.NewForConfigOrDie(env.RestClientConfig)
 	env.APIExtensionClient = apiextensionsclientset.NewForConfigOrDie(env.RestClientConfig)
 	env.Ctx = context.Background()
 	env.Scheme = runtime.NewScheme()
 	env.Log = ctrl.Log.WithName("e2e")
 
-	var err error
+	// Fetching postgres image version.
+	if postgresImage, exist := os.LookupEnv("POSTGRES_IMG"); exist {
+		imageReference := utils.NewReference(postgresImage)
+		postgresImageVersion, err := postgres.GetPostgresVersionFromTag(imageReference.Tag)
+		if err != nil {
+			return nil, err
+		}
+		env.PostgresVersion = postgresImageVersion / 10000
+	}
```

Comment:
What happens if the POSTGRES_IMG variable is not defined? I believe the tests run on the default version, so we should set `env.PostgresVersion` accordingly.

Reply:
Added it now.